src/Features/VisualBasic/Portable/Completion/CompletionProviders/TypeImportCompletionProvider.vb

```diff
@@ -0,0 +1,68 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Completion.Providers
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Extensions.ContextQuery
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports System.Collections.Immutable
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
+
+    Friend NotInheritable Class TypeImportCompletionProvider
+        Inherits AbstractTypeImportCompletionProvider
+
+        Public Sub New(workspace As Workspace)
+            MyBase.New(workspace)
+        End Sub
+
+        Friend Overrides Function IsInsertionTrigger(text As SourceText, characterPosition As Integer, options As OptionSet) As Boolean
+            Return CompletionUtilities.IsDefaultTriggerCharacterOrParen(text, characterPosition, options)
+        End Function
+
+        Protected Overrides Async Function CreateContextAsync(document As Document, position As Integer, cancellationToken As CancellationToken) As Task(Of SyntaxContext)
+            ' Need regular semantic model because we will use it to get imported namepsace symbols.
+            Dim semanticModel = Await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(False)
+            Return Await VisualBasicSyntaxContext.CreateContextAsync(document.Project.Solution.Workspace, semanticModel, position, cancellationToken).ConfigureAwait(False)
+        End Function
+
+        Protected Overrides Function GetNamespacesInScope(location As SyntaxNode, semanticModel As SemanticModel, cancellationToken As CancellationToken) As ImmutableHashSet(Of String)
+
+            ' Names in VB are case insensitive, so case insensitive comparer is required to decide if a namespace is in scope.
+            Dim builder = ImmutableHashSet.CreateBuilder(Of String)(CaseInsensitiveComparison.Comparer)
```

Comment:
about 90% of this logic looks shareable with the C# side.  note: you can use ISyntaxFacts to get the Comparer you need.